### ISSUE TITLE
Invalid TextField announcing invalid state incorrectly on iOS with VO

### ISSUE THREAD

**Issue by philipallen** [adobe/react-spectrum#9119]
### Provide a general summary of the issue here

I hope this isn't a duplicate, I looked through the open issues and couldn't find anything similar. 

Prerequisites:
iOS18.6.2, iPhone 12 Pro, Safari or Chrome or Firefox browser
VoiceOver switched on

Steps to reproduce:
1. Add a valid password here https://react-spectrum.adobe.com/react-aria/forms.html?#realtime-validation 
2. Edit it so it's invalid
3. Edit it again so it's valid again

### 🤔 Expected Behavior?

VO should announce that it's not invalid.

### 😯 Current Behavior

VO announces that it's invalid.

### 🖥️ Steps to Reproduce

Steps to reproduce:
1. Add a valid password here https://react-spectrum.adobe.com/react-aria/forms.html?#realtime-validation 
2. Edit it so it's invalid
3. Edit it again so it's valid again

### Version

Latest on website

### What browsers are you seeing the problem on?

Safari

### If other, please specify.

Chrome, Firefox

### What operating system are you using?

iOS

**Comment by tokapeb** [comment]
https://github.com/user-attachments/assets/8857172f-e463-4344-95f7-d65486998685

**Comment by snowystinger** [comment]
Thanks for reporting, I see this as well
